[Justin Ng Jie Ern] iP

Justinnje opened the pull request:
# Duke
> "Your mind is for having ideas, not holding them." – David Allen [(source)](https://dansilvestre.com/productivity-quotes/)

- text-based
- super easy to use
- for ~~FAST~~ VERY FAST typers

All you need to do is:
1. Download it [here](https://github.com/Justinnje/ip)
2. Double Click it
3. Add your tasks
4. Save your tasks
5. Let Duke manage your tasks for you!

And it is **FREE**!! :wink: :wink:

Features:
- [X] Add Tasks (Todo, Event, or Deadline)
- [X] Mark or Unmark Task
- [X] Find Tasks with relevant Keyword
- [ ] Reminders (coming soon)
<hr>

If you are a Java Programmer, you can use it to practice Java too. Here is the `main` method for you to try.
```java
public static void main(String[] args) {
    run();
}
```

Commit:
Add Gradle support

Justinnje committed:
Updated Duke for Level 1. Parrots command back. only with command "bye" will the program end.

Justinnje committed:
Commit for Level 2. Added add() and list() function. Created another class Control to control how Duke acts

Justinnje committed:
Commit for Level 3. Added a Task class, and a mark and unmark function.

Justinnje committed:
Created 3 other classes ToDo, Event , Deadline to help with inheritance. only allow todo, event and deadline task events from happening.

Justinnje committed:
Changed the location of isTaskChecked()

Justinnje committed:
cleaned up the Main file.

Justinnje committed:
Added 2 exception class to catch exceptions. Cleaned to code a little.

Justinnje committed:
Added a new feature delete. Further improved the expcetion checking - catch index that are out of range array

Justinnje committed:
Cleaned the code a little

Justinnje committed:
Removing commented code

Justinnje committed:
Inconsistent "From Duke: " present in some commands, while absent from some
Lack Text UI testing
General print statements for caught errors

Need to change to have consistent in the way Duke responds to commands.
To allow Testing of the program
More specific to allow users to know why their inputted command is wrong

Right after the scanner scans for next line, Have a print statement. "From Duke"
Changed the input.txt file as well as the EXPECT.TXT file to ensure that ACTUAL.TXT will be equal to EXPECTED.TXT when input.txt in fed into the bash

Justinnje committed:
Create Save Function for the Program

To allow users to come back to where they saved.

To be commited and pushed to another branch

Justinnje committed:
Fix bug from previous commit
Add a save method to allow users another method of saving the tasks into their local device.

Program can now read the file for  users who already have the file in their local device. Gives users a sense of continuity
A Ctrl S function for users. Allows user convenient saves of the tasks

This should be into branch-Level-7

Justinnje committed:
commiting duke.txt

Justinnje committed:
Add  function for program to read time as actual time
Non valid time will throw an error
this is done in the Time class

this should be for branch-Level 8

Justinnje committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Control.java
#	src/main/java/Duke.java

Justinnje committed:
Update load function when loading previously saved file into the program.
Delete unnecessary comments

Load function was not working probably due to merging of 2 branches
Created a helper function to aid in converting dates into the correct format for LocalDateTime class to read String

Justinnje committed:
Create more OOP in the program.

Create 2 classes Iu and Storage
Iu will be the class that handles terminal interactions with user
Storage will handle saving, loading files etc

Justinnje committed:
Refactor Control to TaskList
Storage now controls all Saving and Loading from file
Ui now controls all UI interactions

To improve OOP

Justinnje committed:
Fix bug with Storage class
Move starting line to the main Duke class

Problem: When there is more than 9 items in the previously saved file. The function is unable to detect whethere the item is marked or not.

Justinnje committed:
Parser created

Used to help parse into data by the user

Justinnje committed:
Package for program

Packaged the program accordingly

Justinnje committed:
Add duke.txt file which is the saved file into gitignore.

Since this is a temp file that should be create locally, there is no need to push it

Justinnje committed:
Add JUnit Tests

To help test individual classes in my program

Justinnje committed:
Add Manifest file

for JAR

Justinnje committed:
JavaDocs added for some files

To help document the java file into a html format

This should be in branch-A-JavaDoc branch.

Justinnje committed:
More javadocs are added to help with html documentation

This should be in the branch-A-JavaDoc

Justinnje committed:
Apply Coding Standard.

Changed the format for switch statements
Changed  constants to be all UPPER_CASE

Justinnje committed:
Add find command.
Add help command.

'find keyword' command helps users to search through Duke and find all tasks with a keyword.
'help' command will print out a manual of all the possible commands and their usages.

This should be in branch-Level-9

Justinnje committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/duke/Storage.java
#	src/main/java/duke/Ui.java

Justinnje committed:
Merge branch 'branch-Level-9'

Justinnje committed:
Add javadoc for 1 constant
Change 1 line from help method

seanlaiys commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,102 @@
+    /**
+     * Method to run Duke.
+     */
+    public void uiHandler() {
```
Would using a verb for your method be better?

Justinnje replied:
Changed to run() instead. Thanks

seanlaiys commented on src/main/java/duke/file_management/CreateFile.java:
```diff
@@ -0,0 +1,50 @@
+ *
+ * @author Justin Ng Jie Ern
+ */
+public class CreateFile {
```
Would using a noun for naming class be better?

Justinnje replied:
As there is a Java API called FileWriter, I had to name my file writer class WriteFile. As i wanted some consistency between this 2 classes, I named this class CreateFile. 

Would welcome your suggestions!

seanlaiys commented on src/main/java/duke/file_management/WriteFile.java:
```diff
@@ -0,0 +1,49 @@
+ *
+ * @author Justin Ng Jie Ern
+ */
+public class WriteFile {
```
Similar to "CreateFile" would using a noun for naming class be better?

seanlaiys reviewed (commented):
Overall great job on naming and code quality! Just a few minor things to address.

Justinnje committed:
Change name of uiHandler to run. Verbs should be used
Edit help function

To follow the Coding Naming Standards
Help function was edited to help viewability

Justinnje committed:
Change the arrangement of the Main Method

Justinnje committed:
Merge branch 'add-gradle-support' of https://github.com/Justinnje/ip

Justinnje committed:
Rename filemanagement folder
Compile to Checkstyle
Added Gradle though buggy still
Added CheckStyle

Justinnje committed:
Change code to match CheckStyle
Gradle is now working

Rename exception folder to exceptions
Rename task folder to tasks

Gradle was not working as suppresssions.xml was not in the checkstyle folder

Justinnje committed:
Added GUI gradle dependencies

Justinnje committed:
Add files for GUI

riakhaitan commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,117 @@
+            System.out.println("Month does not exist");
+
+        }// YYYY-MM-DD HH:MM
+        return year + "-" + monthNum + "-" + day + " " + time;
```
Great job naming the variables, very easy to understand

riakhaitan commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,82 @@
+     * Getter for boolean on whether task is marked or not.
+     * @return Boolean of isChecked
+     */
+    public boolean getChecked() {
```
Does not sound like a boolean function, consider changing the name.

riakhaitan commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,82 @@
+     *
+     * @return "X" if the task is marked. " " if the task is unmarked.
+     */
+    public String isTaskCheck() {
```
The name isTaskCheck can be a bit confusing, isTaskCompleted/isTaskChecked would be a better and more descriptive alternative.

riakhaitan reviewed (commented):
Overall, coding standard is followed very well.

Justinnje committed:
Fix GUI.

Now has the function of the previous Text UI.

Problem: when the list of tasks exceeds the size of the DialogBox, the exceeding part gets cut off.

Justinnje committed:
Fix bye bug

bye command will now close the window.

Justinnje committed:
Change the size of the window

Justinnje committed:
edit GUI

Justinnje committed:
Update Java Docs

Justinnje committed:
Change the photo for DaDuke and DaUser

Personalised Duke

Justinnje committed:
Change mainClassName in build.gradle

Currently, the mainClassName is indicated as duke.Duke

This needs to be changed as the JavaFX feature is now added to the
program. After adding JavaFX to Duke, the new Main class of the
code is the Launcher class.

Changed the mainClassName indicated to duke.Launcher. This is to
generate a new fat JAR file with JavaFX included in the JAR file.

Justinnje committed:
Add Assertions to program

The code did not have any assertions. The assumptions about the
program state should be checked and verified at runtime.

Additional checking helps to ensure that code does not have any
errors.

Add to specific methods where there is a clear assumption about
the value of a certain variable.

There are clear assumptions here that should be checked.

Justinnje committed:
Update Code Quality

Some parts of the code did not follow the java coding standard.
Some parts of the code had methods that exceed 60LoC.
Event and deadline tasks did not have a specific check if "/at"
and "/by" were present respectively in the input.

To follow the coding standard. To make the code more readable and
easily more understandable.
This needs to change so that users have a better understand of what
is missing from their inputs.

Break up long codes into smaller chunk.
Add Exception check in both parseEvent and parseDeadline methods.

Using an exception ensures that the error can be found and picked
out.

Justinnje committed:
Merge pull request #2 from Justinnje/branch-A-Assertions

Add Assertions to program

Justinnje committed:
Merge branch 'branch-A-CodeQuality'

# Conflicts:
#	src/main/java/duke/Parser.java

Justinnje committed:
Add Help command to Duke

Users need to know all the commands of the program in order to use
the application. New Users will be at a lose.

First time users of Duke need a way to seek help whenever they are
stuck. Even experienced users may need a reminder on how to use
Duke.

Help command is added. String of all commands in Duke and what they
are used for.

Just a manual for Users to refer to whenever needed.

Justinnje committed:
Change name from Duke to Michael

Justinnje committed:
Update gitignore

Justinnje committed:
Improve GUI

Added a delay for the bye command
Dialog Box is expandable

Justinnje committed:
Reorganize code

Added DialogBox class and MainWindow class into a gui package.
More intuitive understanding of the code.
Changed the help function to include details

Easy location to find code.
Give users an example of the command to aid learning.

Justinnje committed:
Reorganize Code

Justinnje committed:
Add Title Window Image

Justinnje committed:
Change response

Justinnje committed:
Update README.md

Created a UserGuide for users to have a look at to learn how to use
the application. The details of all the features are added and
elaborated on.

Justinnje committed:
Update README.md

Justinnje committed:
Update README.md

Justinnje committed:
Set theme jekyll-theme-cayman

Justinnje committed:
Update README.md

Justinnje committed:
Merge branch 'master' of https://github.com/Justinnje/ip

Justinnje committed:
Update GUI interface

Justinnje committed:
Update screenshot of UI

Justinnje committed:
Update TitleImg

Justinnje committed:
Change TitleImg

Justinnje committed:
Edit README.md

Added a way for users to jump to the usage portion from the
features section of the readme. Allows for Easy Maneuvering
in the readme page.
